client/src/components/Related/RelatedProducts.jsx

```diff
@@ -1,20 +1,31 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import ProductCard from './ProductCard.jsx';
 
-
-
 const RelatedProducts = (props) => {
   const [position, setPosition] = useState(0);
   const componentName = 'RelatedProducts';
+  const [modalShowing, setModalShowing] = useState(false);
+  const [currentProduct, setCurrentProduct] = useState({});
+  const [comparisonProduct, setComparisonProduct] = useState({});
+
+  useEffect(() => {
+    setCurrentProduct(props.currentProduct);
+  }, [props.currentProduct]);
+
+  const getFeatures = () => {
+    return [...currentProduct.features, ...comparisonProduct.features]
+      .filter((v, i, a)=>a.findIndex(v2=>(v.feature === v2.feature && v.value === v2.value)) === i);
```

Comment:
This looks really elegant, but I'm not entirely sure what it's doing. If you have time to refactor, it might help to rename the letter variables to make it more clear what is happening.